packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -58,8 +58,16 @@ export default function createMiddleware(config: MiddlewareConfig) {
     const hasUnknownHost = configWithDefaults.domains != null && !domain;
 
     function getResponseInit() {
-      // Nothing yet
-      return undefined;
+      if (hasOutdatedCookie) {
+        request.headers.set(HEADER_LOCALE_NAME, locale);
```

Comment:
This is not necessary for the current implementation in the main branch. The header is the workaround that is currently used in the RSC branch to "send" the matched locale to Server Components. It's also the reason why SSG doesn't work at this point in the RSC branch, I hope to get rid of this once ServerContext works well.

In the main branch we currently fully support SSG and don't need this workaround, therefore we shouldn't introduce it here.

Can you remove this?

Reply:
@amannn Sorry, my bad! Sure I will remove it.

Reply:
No worries! This is unfortunately a bit convoluted currently.

Btw. I'm currently in process of modernizing the code base a bit, the main branch now uses pnpm/turbo. Feel free to rebase to get a performance boost during development! The [contributors guide](https://github.com/amannn/next-intl/blob/main/CONTRIBUTORS.md) is updated as well.

Next up is modernizing the test runner and build system …